test/Microsoft.TestPlatform.AcceptanceTests/Extension/CompatibilityRowsBuilder.cs

```diff
@@ -149,7 +149,17 @@ public List<RunnerInfo> CreateData()
             throw new InvalidOperationException("There were no rows that matched the specified criteria.");
         }
 
-        return distinctRows.Values.ToList();
+        var allRows = distinctRows.Values.ToList();
+
+        return JustRow == null ? allRows : new List<RunnerInfo> { allRows[JustRow.Value] };
+    }
+
+    private static SemanticVersion ParseAndPatchSemanticVersion(string? version)
+    {
+        // Our developer version is 17.2.0-dev, but we release few preview, that are named 17.2.0-preview or 17.2.0-release, yet we still
+        // want 17.2.0-dev to be considered the latest version. So we patch it.
+        var v = version != null && version.EndsWith("-dev") ? version?.Substring(0, version.Length - 4) + "-ZZZZZZZZZZ" : version;
```

Comment:
```suggestion
        var v = version?.EndsWith("-dev") ? version.Substring(0, version.Length - 4) + "-ZZZZZZZZZZ" : version;
```

Reply:
This won't compile, bool? is not convertible to bool by default. I would have to do your change + ?? false before the ternary ? which makes it worse :)

Reply:
Oh yes I am stupid.

You can do:
```suggestion
        var v = version?.EndsWith("-dev") == true ? version.Substring(0, version.Length - 4) + "-ZZZZZZZZZZ" : version;
```

If you don't, please at least remove the `?.` access in the `Substring`, I had to read it twice to check why we were null checking.